### PR TITLE
[Telink] Update apps and configs & Update Telink image to 0.6.27

### PR DESCRIPTION
**Pull request by s07641069**
**Change overview**

- Rework flash partitions & use swap move algorithm
- Removed timer to BLE disconnect
- Use latest Telink mcuboot changes
- Enable reboot system when fault happened
- Rework system status LED loop
- Made the Factory-Reset Button more user friendly
- Add contact sensor app example
- Fix device re-pairing
- Enable NVS lookup cache
- Update Telink image to 0.6.27

**Testing**

Tested manually with chip-tool.

Steps:

- Run: $ chip-tool pairing ble-thread <...>
- Wait till success
- Run: $ chip-tool onoff on ${NODE_ID }1
- Wait till success
- Run: $ chip-tool onoff off ${NODE_ID }1
- Wait till success

### CODE REVIEW

**Comment by s07641069** [comment]
@andy31415 PR blocked by docker image 0.6.27 build.
After docker image will be available need to retrigger Tests (merge master branch).

File: examples/all-clusters-app/telink/src/AppTask.cpp
Comment (bzbarsky-apple) [inline examples/all-clusters-app/telink/src/AppTask.cpp:118]: This does not look right.

If a commissioner deletes a single existing fabric over PASE during commissioning, before it has done AddNOC, doing a reset does not seem like the right behavior.  Is it?

Comment (andriy-bilynskyy) [reply examples/all-clusters-app/telink/src/AppTask.cpp:118]: Could you explain the scenario to reproduce your situation?
During pairing I've never seen removing last fabric.
In our case this pease of code appears only when pairing is removed. So if latest pairing is removed we are doing factory reset.

Comment (bzbarsky-apple) [reply examples/all-clusters-app/telink/src/AppTask.cpp:118]: > Could you explain the scenario to reproduce your situation?

Say a device is commissioned into a single fabric.  That fabric opens a commissioning window.  A new commissioner connects via PASE, and before it does AddNOC removes the existing fabric.  Why, that's up to it; it just does.  Should that close the PASE session immediately and reset the device?

I just looked more carefully at the spec, and at least the "should it close the PASE session?" answer seems to be "yes".  So doing a reset here is probably fine.

**Comment by andy31415** [comment]
@s07641069 this also seems to need conflict resolution now.

**Comment by s07641069** [comment]
@andy31415 After merge of https://github.com/project-chip/connectedhomeip/pull/24213 need to retest this changes and resolve conflicts again to avoid functionality crash.

**Comment by s07641069** [comment]
@andy31415 @bzbarsky-apple All merge conflicts fixed and PR ready for review/merge.

**Review by Damian-Nordic — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**
